R/get-path-from-object.R

```diff
@@ -158,45 +158,221 @@ get_model_id.bbi_model <- function(.mod) {
 
 #' Get path to data file
 #'
-#' Helper to extract the path to the input data from a
-#' `bbi_{.model_type}_summary` or `bbi_{.model_type}_model` object. Note that
-#' **this function only works on models that have been successfully run**
-#' because the path to the data is constructed during model submission.
-#' @param .mod Model to use, either a `bbi_{.model_type}_model` or
-#' `bbi_{.model_type}_summary` object.
+#' Helper to extract the path to the input data from a bbi object.
+#'
+#' @param .bbi_object The object to query. Could be
+#' a `bbi_{.model_type}_model` object,
+#' `bbi_{.model_type}_summary` object,
+#' or a tibble of class `bbi_log_df`.
+#' @param .check_exists If `TRUE`, the default, will throw an error if the file
+#'  does not exist
+#' @param pull_from_config If `TRUE`, pull the data path from `bbi.json` created
+#'  during model submission, and compare to the path defined in the control stream.
 #' @param ... Arguments passed through to methods. (Currently none, but may have
 #'   some modifier arguments other model types in the future.)
-#' @importFrom fs path_rel
+#'
+#' @details
+#'
+#' ### Notes on model extensions:
+#' `bbi` has different handling for `.mod` vs `.ctl` extensions for `NONMEM`
+#' models. `bbi` was designed to mimic `PsN` behavior. As such, the control
+#' stream file is first copied to a subdirectory before executing the model. The
+#' difference is that on copy `bbi` adjusts a relative data path for `.mod`
+#' files, adding on a `"../"`. Or, said from the user perspective, writers of
+#' `.ctl` files are expected to specify a relative path for the data file
+#' **one level deeper** than the control stream they're editing.
```

Comment:
I like @kyleam 's wording here (and I agree that we don't need the link to bbi code or docs).

I also think putting this in `?submit_model` makes more sense, and that the `?get_data_path` docs should have something like `See "Notes on NONMEM model extensions" in [submit_model()] documentation for more information on handling of data paths in control stream files.`

What do you think about that @barrettk ?

Reply:
I think it would be good to have this information pop up sooner rather than later, which definitely points to `submit_model`. However given the relevance of the information, I think it belongs more with `get_data_path()`. I haven't looked into that `@template` thing @kyleam mentioned yet, but I would lean towards being more verbose in the `get_data_path` helper docs, while mentioning it/linking to the `get_data_path` docs within `submit_model`.

Dont have a strong opinion here if you guys disagree with that logic. I definitely think this information should get exposed *before* `get_data_path`

Reply:
FYI I ended up opting for the `@template` option. I think that's a good practice to start enforcing for documentation we may want to include in multiple places. I followed what `plyr` did [here](https://github.com/hadley/plyr/tree/main/man-roxygen) and [here](https://github.com/hadley/plyr/blob/main/R/aaply.r#L16-L18).